### ISSUE TITLE
Generating data for training

### ISSUE THREAD

**Issue by albertmend** [zakharos/DPOD#2]
Hello,

I am working on my Masters thesis about pose estimation and I am trying to replicate your paper's results. For this purpose, I am trying to generate the files needed for training the network. Based on the original LineMOD ply files, I am trying to generate a new file with one-to-one correspondence between UV values and xyz data. Does this need to be a spherical/cylindrical projection?
Also, could you give any advice on how to generate the files:
synth_XXXX_corr.png         
synth_XXXX_dpt.png          
synth_XXXX_dpt_vis.png
synth_XXXX_img.png
synth_XXXX_norm.png?
I think I can obtain the _corr file by projecting the UV-mapped model over a black background and the _img file projecting the original model, but I don't understand how to obtain the rest of the files.
Are you planning to release the pose refiner code, as well?
The pretrained network weights work fine for each object, but how can I train for multiple objects?
Do you recommend any SfM software for generating custom object models that can be used for training DPOD?

Thank you in advance,

Dennis

**Comment by zakharos** [comment]
Hi Dennis,

- The easiest way to produce correspondence models for training (by projecting UV maps on them) is to use Blender. You simply need to generate a texture and apply it to the 3D model of interest (similar to [link](https://docs.blender.org/manual/en/2.79/editors/uv_image/uv/editing/unwrapping/mapping_types.html#unwrap)). Once the correspondence model is generated, you could use it to produce GT correspondences as described in the paper. You could also use other projection types depending on the complexity of the object geometry to achieve better results. I would however recommend to use the normalized object coordinates described in the supplementary material, section A.7. (UVW Mapping), instead. This way you don't have to deal with any projections.
- To generate the patches, you could use any of the available renderers. In this particular case, we used our own renderer: [https://github.com/zakharos/nocs-renderer](https://github.com/zakharos/nocs-renderer). It is currently set to render UVW correspondences, but if you have your own textured UV correspondence models, then you'd just need to collect the RGB output instead.
- We currently have no plans for releasing the pose refiner code.
- When extending the pipeline to multiple object training, our correspondence branch is class agnostic, the mask head decides for the class.
- As for generating custom models, for [HomebrewedDB](https://arxiv.org/pdf/1904.03167.pdf) we used the Artec Eva scanner together with the provided software. It is quite expensive, but provides very high quality models.

**Comment by albertmend** [comment]
Thank you!
